resident/resident-service/src/test/java/io/mosip/resident/test/util/EncryptorUtilTest.java

```diff
@@ -0,0 +1,121 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.*;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.springframework.core.env.Environment;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.*;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+/**
+ * @author Abubacker Siddik
+ */
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore({"com.sun.org.apache.xerces.*", "javax.xml.*", "org.xml.*", "javax.management.*"})
+@PrepareForTest({DateUtils.class})
+public class EncryptorUtilTest {
+
+    @InjectMocks
+    private EncryptorUtil encryptorUtil;
+
+    @Mock
+    private ObjectMapper objectMapper;
+
+    @Mock
+    private Environment env;
+
+    @Mock
+    private ResidentServiceRestClient restClientService;
+
+    @Captor
+    ArgumentCaptor<RequestWrapper<CryptomanagerRequestDto>> requestCaptor;
+
+    @Captor
+    ArgumentCaptor<String> stringCaptor;
+
+    private LocalDateTime localDateTime;
+    private String encryptAPIUrl;
+
+    @Before
+    public void setUp() throws Exception {
+        encryptAPIUrl = "http://kernel-keymanager-service/v1/keymanager/encrypt";
+
+        PowerMockito.mockStatic(DateUtils.class);
+
+        localDateTime = DateUtils.getUTCCurrentDateTime();
+        when(DateUtils.getUTCCurrentDateTime()).thenReturn(localDateTime);
+    }
+
+    @Test
+    public void encryptTest() throws Exception {
+        DecryptResponseDto decryptResponseDto = new DecryptResponseDto();
+        decryptResponseDto.setData("ew0KCSJpZGVudGl0eSI6IHsNCgkJImRhdGVPZkJpcnRoIjogIjE5NzgvMDEvMDEiLA0KCQkiSURTY2hlbWFWZXJzaW9uIjogMC4xLA0KCQkiVUlOIjogIjkzNDIzNjEwMzgiDQoJfQ0KfQ");
+        io.mosip.resident.dto.ResponseWrapper<DecryptResponseDto> responseWrapper = new io.mosip.resident.dto.ResponseWrapper<>();
+        responseWrapper.setResponse(decryptResponseDto);
+        String encryptedJsonString = "{ data: ew0KCSJpZGVudGl0eSI6IHsNCgkJImRhdGVPZkJpcnRoIjogIjE5NzgvMDEvMDEiLA0KCQkiSURTY2hlbWFWZXJzaW9uIjogMC4xLA0KCQkiVUlOIjogIjkzNDIzNjEwMzgiDQoJfQ0KfQ }";
+
+        when(env.getProperty(Mockito.anyString())).thenReturn(encryptAPIUrl);
+        when(restClientService.postApi(Mockito.anyString(), Mockito.any(), Mockito.any(), Mockito.any(Class.class))).thenReturn(responseWrapper);
+        when(objectMapper.readValue(Mockito.anyString(), Mockito.any(Class.class))).thenReturn(decryptResponseDto);
+        when(objectMapper.writeValueAsString(responseWrapper.getResponse())).thenReturn(encryptedJsonString);
+
+        byte[] bytes = "Encrypt String".getBytes();
+        String refId = "CenterID_MachineID";
+        encryptorUtil.encrypt(bytes, refId);
```

Comment:
this method seems to be returning the entrypted string, can you do a assert on the return value ?

Reply:
As discussed, i am not asserting the returned value.